protocols/gossipsub/src/lib.rs

```diff
@@ -156,18 +156,53 @@ mod types;
 
 mod rpc_proto;
 
-pub use self::behaviour::{Gossipsub, GossipsubEvent, MessageAuthenticity};
+pub use self::behaviour::{Behaviour, Event, MessageAuthenticity};
 pub use self::transform::{DataTransform, IdentityTransform};
 
-pub use self::config::{GossipsubConfig, GossipsubConfigBuilder, GossipsubVersion, ValidationMode};
 pub use self::peer_score::{
     score_parameter_decay, score_parameter_decay_with_base, PeerScoreParams, PeerScoreThresholds,
     TopicScoreParams,
 };
 pub use self::topic::{Hasher, Topic, TopicHash};
-pub use self::types::{
-    FastMessageId, GossipsubMessage, GossipsubRpc, MessageAcceptance, MessageId,
-    RawGossipsubMessage,
-};
+pub use self::types::{FastMessageId, Message, MessageAcceptance, MessageId, RawMessage, Rpc};
+
+pub use self::config::{Builder, Config, ValidationMode, Version};
+
+#[deprecated(
+    since = "0.44.0",
+    note = "Use re-exports that omit `Gossipsub` prefix, i.e. `libp2p::gossipsub::Config"
+)]
+pub type GossipsubConfig = config::Config;
+
+#[deprecated(
+    since = "0.44.0",
+    note = "Use re-exports that omit `Gossipsub` prefix, i.e. `libp2p::gossipsub::Message"
+)]
+pub type GossipsubMessage = Message;
+
+#[deprecated(
+    since = "0.44.0",
+    note = "Use re-exports that omit `Gossipsub` prefix, i.e. `libp2p::gossipsub::Rpc"
+)]
+pub type GossipsubRpc = Rpc;
+
+#[deprecated(
+    since = "0.44.0",
+    note = "Use re-exports that omit `Gossipsub` infix, i.e. `libp2p::gossipsub::RawMessage"
+)]
+pub type RawGossipsubMessage = RawMessage;
+
+#[deprecated(
+    since = "0.44.0",
+    note = "Use re-exports that omit `Gossipsub` prefix, i.e. `libp2p::gossipsub::Builder"
+)]
+pub type GossipsubBuilder = config::Builder;
```

Comment:
```suggestion
#[deprecated(
    since = "0.44.0",
    note = "Use re-exports that omit `Gossipsub` prefix, i.e. `libp2p::gossipsub::Builder"
)]
pub type GossipsubConfigBuilder = config::Builder;
```

That is what the type used to be called.

Reply:
I was actually mistaken with what I wrote in https://github.com/libp2p/rust-libp2p/pull/3303#discussion_r1064236220, sorry to send you on a wild goose chase here.

Reply:
No worries at all :smile: